### PR TITLE
renesas-ra/machine_pin.c:  Support drive keyword and fix GPIO pin setting.

### PR DESCRIPTION
**Pull request by TakeoTakahashi2020**
* Support drive= keyword argument for Pin module.
* Fix trigger keyword check for Pin() and Pin.init().
* Fix GPIO pin setting for Pin() and Pin.init().
* Update Pin test for common.
* Update document to add driver= keyword argument description.

### CODE REVIEW

File: ports/renesas-ra/machine_pin.c [hunk 1]
```diff
@@ -159,29 +161,50 @@ STATIC mp_obj_t machine_pin_obj_init_helper(const machine_pin_obj_t *self, size_
...
     }
 
     // get pull mode
-    uint pull = 0;
-    if (args[1].u_obj != mp_const_none) {
-        pull = mp_obj_get_int(args[1].u_obj);
+    uint32_t pull;
+    if (args[ARG_pull].u_obj != mp_const_none) {
+        pull = mp_obj_get_int(args[ARG_pull].u_obj);
+        if (!IS_GPIO_MODE(pull)) {
```
Comment (dpgeorge) [inline ports/renesas-ra/machine_pin.c:178]: this should be `IS_GPIO_PULL`

Comment (TakeoTakahashi2020) [reply ports/renesas-ra/machine_pin.c]: Thank you for your review.  I fixed and updated.

**Comment by dpgeorge** [comment]
Thank you for this.  It has been rebased and merged in 621bff8557c7a7e35cdc9f6c6de0e6c912d94202 through af100b70290d20d7e9c3e7d49713e3916b6f05b6 (with some very minor changes to the exception messages when something is not supported, and minor doc formatting change to use `*` around keyword argument names).

**Comment by TakeoTakahashi2020** [comment]
Thank you very much.

**Pull request closed without merging**